build-logic-settings/build-environment/src/main/kotlin/gradlebuild/basics/BuildEnvironmentService.kt

```diff
@@ -14,15 +14,26 @@
  * limitations under the License.
  */
 
+/**
+ * TODO: Remove once with Gradle 9.0, used so org.gradle.kotlin.dsl.* is kept
+ */
+@file:Suppress("UnusedImport")
+
 package gradlebuild.basics
 
 import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.provider.Provider
 import org.gradle.api.provider.ProviderFactory
 import org.gradle.api.services.BuildService
 import org.gradle.api.services.BuildServiceParameters
+import org.gradle.internal.deprecation.DeprecationLogger
 import org.gradle.internal.os.OperatingSystem
 import javax.inject.Inject
+/**
+ * Used to import assign for Gradle 9.0
+ * TODO: Remove once with Gradle 9.0
+ */
+import org.gradle.kotlin.dsl.*
```

Comment:
🎉 Nice trick! :) I added this issue with an item to remind us to come back to these comments: https://github.com/gradle/gradle/issues/31027